### PR TITLE
API Request Query and Webpack Configuration enhancements

### PR DESCRIPTION
**Pull request by drtyh2o**
- More expressive type for API query parameters
- Use builder pattern for webpack configuration to simplify syntax

### CODE REVIEW

File: packages/mds-api-server/index.ts [hunk 1]
```diff
@@ -9,7 +9,18 @@ import { Params, ParamsDictionary } from 'express-serve-static-core'
 
 export type ApiRequest<P extends Params = ParamsDictionary> = express.Request<P>
 
-export type ApiQuery<Q extends string> = { query: Partial<{ [P in Q]: string | string[] }> }
+export type ApiQuery<Q1 extends string, Q2 extends string[] = never> = {
+  query: Partial<
+    {
+      [P in Q1]: string
```
Comment (wesleyfsmith) [inline packages/mds-api-server/index.ts:15]: This syntax is new to me, can you give a little explanation?

File: packages/mds-api-server/index.ts [hunk 2]
```diff
@@ -9,7 +9,18 @@ import { Params, ParamsDictionary } from 'express-serve-static-core'
...
+    }
+  > &
+    Partial<
+      {
+        [P in Q2[number]]: string | string[]
+      }
+    > & { [x: string]: never }
```
Comment (janedotx) [inline packages/mds-api-server/index.ts:22]: What's the purpose of this `never` type in `ApiQuery`?

File: packages/mds-webpack-config/index.ts [hunk 3]
```diff
@@ -118,16 +118,22 @@ const MergeConfigurations = (config: CustomConfiguration, name: string, path?: s
...
-  UsingCustomConfig: (config: CustomConfiguration) => MergeConfigurations(config, name, path)
-})
+class WebpackConfigurationBuilder {
+  private name: string
+
+  constructor(private path: string, name?: string) {
+    this.name = name || parse(path).name
+  }
+
+  public UsingDefaultConfig() {
```
Comment (wesleyfsmith) [inline packages/mds-webpack-config/index.ts:128]: What's the distinction between a default config and a custom config?

File: packages/mds-api-server/index.ts (hunk 2 above)
Comment (drtyh2o) [reply packages/mds-api-server/index.ts:22]: It forces you to specify all of the expected query string parameters. It's best to illustrate with an example:
```ts
const req: ApiRequest & ApiQuery<'one' | 'only', ['many', 'more']> = ...
const { one, only, many, more, idk } = req.query
```
Gets typed as 
```ts
{
   one: string | undefined
   only: string | undefined
   many: string | string[] | undefined
   more: string | string[] | undefined
   idk: never
}
```
So, basically, any property other than those in the type declaration gets typed as `never` meaning you really can't do anything with it.

File: packages/mds-api-server/index.ts (hunk 1 above)
Comment (drtyh2o) [reply packages/mds-api-server/index.ts:15]: Basically, there is an optional `string` property on the object for every name in Q1 and an optional string or array of strings property on the object for every name in Q2. Any property that isn't specified can not be used. Forces type safe access.

File: packages/mds-webpack-config/index.ts (hunk 3 above)
Comment (drtyh2o) [reply packages/mds-webpack-config/index.ts:128]: A custom config allows overriding the standard webpack config with additional settings, plugins, etc. that are specifically required by the bundle being generated.

**Pull request merged**